--- /home/aditya/Documents/Autonomous-Garden-Rover/firmware/odesc-v42/source/Firmware/MotorControl/main.cpp	2026-08-10 22:26:50.063584801 -0400
+++ Firmware/MotorControl/main.cpp	2026-08-10 23:17:59.848113890 -0400
@@ -298,6 +298,10 @@
  * It should finish as quickly as possible.
  */
 void ODrive::do_fast_checks() {
+    if (vbus_voltage_external_fault) {
+        disarm_with_error(ERROR_EXTERNAL_VBUS_INVALID);
+        return;
+    }
     if (!(vbus_voltage >= config_.dc_bus_undervoltage_trip_level))
         disarm_with_error(ERROR_DC_BUS_UNDER_VOLTAGE);
     if (!(vbus_voltage <= config_.dc_bus_overvoltage_trip_level))
